velox/expression/CastExpr-inl.h

```diff
@@ -126,6 +126,23 @@ StringView convertToStringView(
   return StringView(startPosition, writePosition - startPosition);
 }
 
+inline bool hasWhitespace(const StringView& value) {
+  if (value.size() > 0) {
+    char first = value.data()[0];
+    if (first == ' ' || first == '\t' || first == '\n' || first == '\r' ||
```

Comment:
Are these different from asciiWhitespaceCodes in velox/functions/lib/string/StringImpl.h? Where does this list come from? Would you add a comment?

Reply:
Got the list of `std::isspace()`, https://en.cppreference.com/w/cpp/string/byte/isspace. Also referred to `characterIsSpace()` in TimestampConversion. I ran benchmark and found implementing by ourselves is slightly faster than calling std::isspace(), consistently. So copying here, ours [P863143778](https://github.com/phabricator/paste/view/P863143778) vs std::isspace() [P863136802](https://github.com/phabricator/paste/view/P863136802).

Reply:
@gggrace14 Can you reference Presto code you are matching here? Just want to make sure this list is exactly what is used in Presto.

Reply:
Hi @mbasmanova, Presto code doesn't have a code match here. Presto impl points to https://github.com/prestodb/presto/blob/04138bc26a4fc801630fe936463d89d071d00c0a/presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java#L182, then it is all of the behavior of **`Long.parseLong()`** to not accept leading and trailing whitespaces.  Java doc is https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#parseLong-java.lang.String-.

As you might know, velox impl delegates to folly::to<>(). I referred to the folly::to() impl and saw folly explicitly accepts and trims the leading and trailing whitespaces, using std::isspace. So my change is kind of to offset folly's behavior by throwing before entering folly.
folly https://github.com/facebook/folly/blob/00b7e630bdbd09e7f1cd173c80c2f52b1e631df6/folly/Conv.h#L204-L206
https://github.com/facebook/folly/blob/00b7e630bdbd09e7f1cd173c80c2f52b1e631df6/folly/Conv.h#L1501-L1504
https://github.com/facebook/folly/blob/00b7e630bdbd09e7f1cd173c80c2f52b1e631df6/folly/Conv.h#L1458-L1466

Reply:
@gggrace14 Got it. Thank you for explaining.